--- /Users/brandon/git/Blocks/Source/Core/blocks/app/services/InstallService.php	2012-08-03 05:39:36.000000000 -0700
+++ /Users/brandon/git/Blocks/Source/BlocksPro/blocks/app/services/InstallService.php	2012-08-03 05:40:56.000000000 -0700
@@ -75,6 +75,7 @@
 			$info->release_date = Blocks::getReleaseDate();
 			$info->site_name = $inputs['sitename'];
 			$info->site_url = $inputs['url'];
+			$info->license_key = $inputs['licensekey'];
 			$info->on = true;
 			$info->save();
 
