--- /Users/brandon/git/Blocks/Source/Core/blocks/app/services/InstallService.php	2012-08-24 16:25:27.000000000 -0700
+++ /Users/brandon/git/Blocks/Source/BlocksPro/blocks/app/services/InstallService.php	2012-08-24 16:46:23.000000000 -0700
@@ -77,9 +77,26 @@
 			$info->site_name = $inputs['sitename'];
 			$info->site_url = $inputs['url'];
 			$info->language = $inputs['language'];
+			$info->license_key = $inputs['licensekey'];
 			$info->on = true;
 			$info->save();
 
+			// Register the email messages
+			$message = blx()->email->registerMessage('verify_email');
+			blx()->email->saveMessageContent($message->id,
+				Blocks::t('verify_email_subject'),
+				Blocks::t('verify_email_body'));
+
+			$message = blx()->email->registerMessage('verify_new_email');
+			blx()->email->saveMessageContent($message->id,
+				Blocks::t('verify_new_email_subject'),
+				Blocks::t('verify_new_email_body'));
+
+			$message = blx()->email->registerMessage('forgot_password');
+			blx()->email->saveMessageContent($message->id,
+				Blocks::t('forgot_password_subject'),
+				Blocks::t('forgot_password_body'));
+
 			// Add the user
 			$user = new User();
 			$user->username   = $inputs['username'];
