--- /Users/brandon/git/Blocks/Source/Core/blocks/app/variables/BlxVariable.php	2012-08-03 23:18:18.000000000 -0700
+++ /Users/brandon/git/Blocks/Source/BlocksPro/blocks/app/variables/BlxVariable.php	2012-08-03 23:18:26.000000000 -0700
@@ -117,6 +117,14 @@
 	}
 
 	/**
+	 * @return RoutesVariable
+	 */
+	public function routes()
+	{
+		return new RoutesVariable();
+	}
+
+	/**
 	 * @return SettingsVariable
 	 */
 	public function settings()
