--- /Users/Brad/Dropbox/GitRepos/Blocks/Source/Core/blocks/app/business/webservices/EtPackage.php	2012-08-03 15:05:54.000000000 -0700
+++ /Users/Brad/Dropbox/GitRepos/Blocks/Source/BlocksPro/blocks/app/business/webservices/EtPackage.php	2012-08-03 15:06:34.000000000 -0700
@@ -12,6 +12,7 @@
 	public $requestTime;
 	public $requestDomain;
 	public $requestPort;
+	public $product;
 	public $data;
 	public $errors = array();
 
@@ -26,6 +27,7 @@
 		$this->url = isset($properties['url']) ? $properties['url'] : null;
 		$this->licenseKey = isset($properties['licenseKey']) ? $properties['licenseKey'] : null;
 		$this->data = isset($properties['data']) ? $properties['data'] : null;
+		$this->product = isset($properties['product']) ? $properties['product'] : null;
 		$this->requestDomain = isset($properties['requestDomain']) ? $properties['requestDomain'] : null;
 		$this->requestIp = isset($properties['requestIp']) ? $properties['requestIp'] : null;
 		$this->requestTime = isset($properties['requestTime']) ? $properties['requestTime'] : null;
