--- /Users/brandon/git/Blocks/Source/Core/blocks/app/controllers/EmailController.php	2012-08-17 16:09:09.000000000 -0700
+++ /Users/brandon/git/Blocks/Source/BlocksPro/blocks/app/controllers/EmailController.php	2012-08-18 09:21:56.000000000 -0700
@@ -9,6 +9,27 @@
 	/**
 	 *
 	 */
+	public function actionSaveMessage()
+	{
+		$this->requirePostRequest();
+		$this->requireAjaxRequest();
+
+		$messageId = blx()->request->getRequiredPost('messageId');
+		$language = blx()->request->getPost('language');
+		$subject = blx()->request->getRequiredPost('subject');
+		$body = blx()->request->getRequiredPost('body');
+
+		$content = blx()->email->saveMessageContent($messageId, $subject, $body, null, $language);
+
+		if ($content->getErrors())
+			$this->returnErrorJson(Blocks::t('There was a problem saving your message.'));
+		else
+			$this->returnJson(array('success' => true));
+	}
+
+	/**
+	 *
+	 */
 	public function actionSendTestEmail()
 	{
 		$user = blx()->accounts->getCurrentUser();
