--- /Users/brandon/git/Blocks/Source/Core/blocks/app/controllers/InstallController.php	2012-08-02 15:51:00.000000000 -0700
+++ /Users/brandon/git/Blocks/Source/BlocksPro/blocks/app/controllers/InstallController.php	2012-08-02 15:51:38.000000000 -0700
@@ -38,6 +38,25 @@
 	}
 
 	/**
+	 * Validates the license key
+	 */
+	public function actionValidateLicensekey()
+	{
+		$this->requirePostRequest();
+		$this->requireAjaxRequest();
+
+		$licenseKey = new InstallLicenseKeyForm();
+		$licenseKey->licensekey = blx()->request->getPost('licensekey');
+
+		if ($licenseKey->validate())
+			$return['validates'] = true;
+		else
+			$return['errors'] = $licenseKey->getErrors();
+
+		$this->returnJson($return);
+	}
+
+	/**
 	 * Validates the user account credentials
 	 */
 	public function actionValidateAccount()
@@ -87,6 +106,7 @@
 		$this->requireAjaxRequest();
 
 		// Run the installer
+		$inputs['licensekey'] = blx()->request->getPost('licensekey');
 		$inputs['username']   = blx()->request->getPost('username');
 		$inputs['email']      = blx()->request->getPost('email');
 		$inputs['password']   = blx()->request->getPost('password');
