--- /Users/brandon/git/Blocks/Source/Core/blocks/app/controllers/SettingsController.php	2012-08-14 16:40:15.000000000 -0700
+++ /Users/brandon/git/Blocks/Source/BlocksPro/blocks/app/controllers/SettingsController.php	2012-08-18 09:24:27.000000000 -0700
@@ -24,12 +24,14 @@
 		$generalSettingsForm = new GeneralSettingsForm();
 		$generalSettingsForm->siteName = blx()->request->getPost('siteName');
 		$generalSettingsForm->siteUrl = blx()->request->getPost('siteUrl');
+		$generalSettingsForm->licenseKey = blx()->request->getPost('licenseKey');
 
 		if ($generalSettingsForm->validate())
 		{
 			$info = Info::model()->find();
 			$info->site_name = $generalSettingsForm->siteName;
 			$info->site_url = $generalSettingsForm->siteUrl;
+			$info->license_key = $generalSettingsForm->licenseKey;
 			$info->save();
 
 			blx()->user->setNotice(Blocks::t('General settings saved.'));
@@ -80,6 +82,7 @@
 			$settings = array('protocol' => $emailSettings->protocol);
 			$settings['emailAddress'] = $emailSettings->emailAddress;
 			$settings['senderName'] = $emailSettings->senderName;
+			$settings['template'] = blx()->request->getPost('template');
 
 			switch ($emailSettings->protocol)
 			{
