--- /Users/Brad/Dropbox/GitRepos/Blocks/Source/Core/blocks/app/blocks.php	2012-07-24 16:44:37.000000000 -0700
+++ /Users/Brad/Dropbox/GitRepos/Blocks/Builds/BlocksPro/Blocks/blocks/app/blocks.php	2012-07-27 10:04:00.000000000 -0700
@@ -1,6 +1,6 @@
 <?php
 
 if (!defined('PHP_VERSION_ID') || PHP_VERSION_ID < 50300)
-	exit('requires PHP 5.3.0 or later, but you&rsquo;re running '.PHP_VERSION.'. Please talk to your host/IT department about upgrading PHP or your server.');
+	exit('Blocks Pro requires PHP 5.3.0 or later, but you&rsquo;re running '.PHP_VERSION.'. Please talk to your host/IT department about upgrading PHP or your server.');
 
 require_once 'blocks_init.php';
