--- /Users/brandon/git/Blocks/Source/Core/blocks/app/business/web/UrlManager.php	2012-08-02 14:08:50.000000000 -0700
+++ /Users/brandon/git/Blocks/Source/BlocksPro/blocks/app/business/web/UrlManager.php	2012-08-02 14:10:17.000000000 -0700
@@ -113,6 +113,16 @@
 				}
 			}
 		}
+		else
+		{
+			// Check the user-defined routes
+			$siteRoutes = blx()->routes->getAllRoutes();
+			foreach ($siteRoutes as $route)
+			{
+				if ($this->_matchRouteInternal($route->url_pattern))
+					return $route->template;
+			}
+		}
 
 		return false;
 	}
